### PR TITLE
Check if assets path is writable on publish (#15850)

### PR DESCRIPTION
**Pull request by aivchen**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15850

### CODE REVIEW

File: framework/web/AssetManager.php [hunk 1]
```diff
@@ -211,8 +211,6 @@ public function init()
         $this->basePath = Yii::getAlias($this->basePath);
         if (!is_dir($this->basePath)) {
```
Comment (alexeevdv) [inline framework/web/AssetManager.php]: What about this check? Shouldn't it be moved too? It seems like there is no reason to check `basePath` if nothing will be published there

Comment (Groonya) [reply framework/web/AssetManager.php]: May be it should be moved, but the issue is about checking $basePath for writability only.

Also I've found one problem in test.
To emulate read only dir I run chmod($path, 0555) in the test. But after that developer can have a problem to switch to another branch. 
Can you suggest something? Thanks.

Comment (alexeevdv) [reply framework/web/AssetManager.php]: You can create this folder in `setUp` and delete it in `tearDown` method

Comment (alexeevdv) [reply framework/web/AssetManager.php]: Or you can `chmod` it back in `tearDown`

Comment (Groonya) [reply framework/web/AssetManager.php]: Done, thanks.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**